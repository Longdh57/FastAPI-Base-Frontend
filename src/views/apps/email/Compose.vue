<template>
  <div>
    <p class="text-center">New Message</p>
    <form>
      <div class="form-row mb-3">
        <label for="to" class="col-2 col-sm-1 col-form-label">To:</label>
        <CCol cols="12" sm="11">
          <input type="email" class="form-control" id="to" placeholder="Type email">
        </CCol>
      </div>
      <div class="form-row mb-3">
        <label for="cc" class="col-2 col-sm-1 col-form-label">CC:</label>
        <CCol cols="12" sm="11">
          <input type="email" class="form-control" id="cc" placeholder="Type email">
        </CCol>
      </div>
      <div class="form-row mb-3">
        <label for="bcc" class="col-2 col-sm-1 col-form-label">BCC:</label>
        <CCol cols="12" sm="11">
          <input type="email" class="form-control" id="bcc" placeholder="Type email">
        </CCol>
      </div>
    </form>
    <CRow>
      <CCol sm="11" class="ml-auto">
        <div class="toolbar" role="toolbar">
          <div class="btn-group mr-1">
            <CButton color="light"><CIcon name="cil-bold"/></CButton>
            <CButton color="light"><CIcon name="cil-italic"/></CButton>
            <CButton color="light"><CIcon name="cil-underline"/></CButton>
          </div>
          <div class="btn-group mr-1">
            <CButton color="light"><CIcon name="cil-align-left"/></CButton>
            <CButton color="light"><CIcon name="cil-align-right"/></CButton>
            <CButton color="light"><CIcon name="cil-align-center"/></CButton>
            <CButton color="light"><CIcon name="cil-justify-center"/></CButton>
          </div>
          <div class="btn-group mr-1">
            <CButton color="light"><CIcon name="cil-indent-increase"/></CButton>
            <CButton color="light"><CIcon name="cil-indent-decrease"/></CButton>
          </div>
          <div class="btn-group mr-1">
            <CButton color="light"><CIcon name="cil-list"/></CButton>
            <CButton color="light"><CIcon name="cil-list-numbered"/></CButton>
          </div>
          <CButton color="light" class="mr-1"><CIcon name="cil-trash"/></CButton>
          <CButton color="light"><CIcon name="cil-paperclip"/></CButton>
          <LabelDropdown/>
        </div>
        <div class="form-group mt-4">
          <textarea 
            class="form-control" 
            id="message" 
            name="body" 
            rows="12" 
            placeholder="Click here to reply"
          ></textarea>
        </div>
        <div class="form-group">
          <CButton class="mr-1" type="submit" color="success">Send</CButton>
          <CButton class="mr-1" type="submit" color="light">Draft</CButton>
          <CButton class="mr-1" type="submit" color="danger">Discard</CButton>
        </div>
      </CCol>
    </CRow>
  </div>
</template>

<script>
import LabelDropdown from './LabelDropdown'

export default {
  name: 'Compose',
  components: {
    LabelDropdown
  }
}
</script>