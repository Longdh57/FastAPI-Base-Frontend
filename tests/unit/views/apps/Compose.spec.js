import {shallowMount, createLocalVue} from '@vue/test-utils';
import VueRouter from 'vue-router'
import CoreuiVue from '@coreui/vue-pro'
import Compose from '@/views/apps/email/Compose'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('Compose.vue', () => {
  it('has a name', () => {
    expect(Compose.name).toMatch('Compose')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Compose, {
      localVue,
      router
    })
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Compose', () => {
    const wrapper = shallowMount(Compose, {
      localVue,
      router
    })
    expect(wrapper.findComponent(Compose)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Compose, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

