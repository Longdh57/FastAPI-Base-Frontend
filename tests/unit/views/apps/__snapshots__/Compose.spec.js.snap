// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compose.vue renders correctly 1`] = `
<div>
  <p
    class="text-center"
  >
    New Message
  </p>
   
  <form>
    <div
      class="form-row mb-3"
    >
      <label
        class="col-2 col-sm-1 col-form-label"
        for="to"
      >
        To:
      </label>
       
      <ccol-stub
        cols="12"
        sm="11"
        tag="div"
      >
        <input
          class="form-control"
          id="to"
          placeholder="Type email"
          type="email"
        />
      </ccol-stub>
    </div>
     
    <div
      class="form-row mb-3"
    >
      <label
        class="col-2 col-sm-1 col-form-label"
        for="cc"
      >
        CC:
      </label>
       
      <ccol-stub
        cols="12"
        sm="11"
        tag="div"
      >
        <input
          class="form-control"
          id="cc"
          placeholder="Type email"
          type="email"
        />
      </ccol-stub>
    </div>
     
    <div
      class="form-row mb-3"
    >
      <label
        class="col-2 col-sm-1 col-form-label"
        for="bcc"
      >
        BCC:
      </label>
       
      <ccol-stub
        cols="12"
        sm="11"
        tag="div"
      >
        <input
          class="form-control"
          id="bcc"
          placeholder="Type email"
          type="email"
        />
      </ccol-stub>
    </div>
  </form>
   
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      class="ml-auto"
      sm="11"
      tag="div"
    >
      <div
        class="toolbar"
        role="toolbar"
      >
        <div
          class="btn-group mr-1"
        >
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-bold"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-italic"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-underline"
            />
          </cbutton-stub>
        </div>
         
        <div
          class="btn-group mr-1"
        >
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-align-left"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-align-right"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-align-center"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-justify-center"
            />
          </cbutton-stub>
        </div>
         
        <div
          class="btn-group mr-1"
        >
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-indent-increase"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-indent-decrease"
            />
          </cbutton-stub>
        </div>
         
        <div
          class="btn-group mr-1"
        >
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-list"
            />
          </cbutton-stub>
           
          <cbutton-stub
            activeclass="router-link-active"
            color="light"
            event="click"
            exactactiveclass="router-link-exact-active"
            routertag="a"
            target="_self"
            type="button"
          >
            <cicon-stub
              name="cil-list-numbered"
            />
          </cbutton-stub>
        </div>
         
        <cbutton-stub
          activeclass="router-link-active"
          class="mr-1"
          color="light"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="button"
        >
          <cicon-stub
            name="cil-trash"
          />
        </cbutton-stub>
         
        <cbutton-stub
          activeclass="router-link-active"
          color="light"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="button"
        >
          <cicon-stub
            name="cil-paperclip"
          />
        </cbutton-stub>
         
        <labeldropdown-stub />
      </div>
       
      <div
        class="form-group mt-4"
      >
        <textarea
          class="form-control"
          id="message"
          name="body"
          placeholder="Click here to reply"
          rows="12"
        />
      </div>
       
      <div
        class="form-group"
      >
        <cbutton-stub
          activeclass="router-link-active"
          class="mr-1"
          color="success"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="submit"
        >
          Send
        </cbutton-stub>
         
        <cbutton-stub
          activeclass="router-link-active"
          class="mr-1"
          color="light"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="submit"
        >
          Draft
        </cbutton-stub>
         
        <cbutton-stub
          activeclass="router-link-active"
          class="mr-1"
          color="danger"
          event="click"
          exactactiveclass="router-link-exact-active"
          routertag="a"
          target="_self"
          type="submit"
        >
          Discard
        </cbutton-stub>
      </div>
    </ccol-stub>
  </crow-stub>
</div>
`;
